tests/scadnano_tests.py

```diff
@@ -91,6 +91,177 @@ def test_strand__loopouts_with_labels_to_json(self) -> None:
         self.assertEqual(1, len(design_from_json.strands))
         self.assertEqual(expected_strand, design_from_json.strands[0])
 
+    def test_strand__extension_3p_from_to(self) -> None:

```

Comment:
Slight critique: I tend to create the empty `Design` in circumstances like this in a `setUp()` "fixture" (as the unit test people call it), and then access that (as `self.design`) to create specific designs to test later. Not a huge deal here since it's only a one-liner to create the design.

Reply:
Good point, I will rewrite the tests I wrote to use `design = self.design_6helix` for consistency with the existing tests